.github/workflows/tests.yml

```diff
@@ -73,69 +74,77 @@ jobs:
+      - name: Cache conda
+        uses: actions/cache@v2
+        with:
+          path: ~/conda_pkgs_dir
+          key: ${{ runner.os }}-conda-${{ env.TIMESTAMP }}
+
+      - name: Setup miniconda
         uses: conda-incubator/setup-miniconda@v2
         with:
-          condarc-file: ci/github/.condarc
+          condarc-file: ./src/ci/github/.condarc
           python-version: ${{ matrix.python-version }}
+          run-post: false  # skip post cleanup
 
-      - name: Install Conda-Canary if canary version
-        if: matrix.conda-version == 'canary'
-        run: |
-          conda install -y -c conda-canary/label/dev conda
-
-      - name: Install default Conda if release version
-        if: matrix.conda-version == 'release'
+      - name: Setup environment
         run: |
-          conda install -y conda
+          conda install -q -y -c defaults \
+            --file ./tests/requirements.txt \
+            --file ./tests/requirements-linux.txt \
+            ${{ env.CONDA_CHANNEL_LABEL }}::conda
+          pip install allure-pytest
+          pip install -e .
 
-      - name: Source Scripts
+      - name: Show info
         run: |
-          source ci/github/install_conda_build_test_deps
-          pip install -e .
-          pip install allure-pytest
           conda info -a
+          conda list --show-channel-urls
 
-      - name: Run Serial tests
-        if: matrix.test-type == 'serial'
-        run: |
-          pytest \
-              --color=yes \
-              -vv \
-              -n 0 \
-              --basetemp "${{ runner.temp }}/${{ matrix.test-type }}" \
-              --cov conda_build \
-              --cov-report xml \
-              -m "serial" tests \
-              ${{ env.pytest-replay }} \
-              --alluredir=allure-results
-
-      - name: Run Parallel tests
-        if: matrix.test-type == 'parallel'
+      - name: Run tests
         run: |
           pytest \
-              --color=yes \
-              -vv \
-              -n auto \
-              --basetemp "${{ runner.temp }}/${{ matrix.test-type }}" \
-              --cov conda_build \
-              --cov-append \
-              --cov-report xml \
-              -m "not serial" tests \
-              ${{ env.pytest-replay }} \
-              --alluredir=allure-results
+            --color=yes \
+            -v \
+            -n "${{ env.PYTEST_NUMPROCESSES }}" \
+            --basetemp "${{ runner.temp }}/${{ matrix.test-type }}" \
+            --cov conda_build \
+            --cov-append \
+            --cov-report xml \
+            --replay-record-dir="${{ env.REPLAY_DIR }}" \
+            --replay-base-name="${{ env.REPLAY_NAME }}" \
+            --alluredir="${{ env.ALLURE_DIR }}" \
+            -m "${{ env.PYTEST_MARKER }}" \
+            ./tests
 
       - name: Tar Allure Results
         if: always()
-        run: tar -zcf allure-results.tar.gz allure-results
+        run: tar -zcf "${{ env.ALLURE_DIR }}.tar.gz" "${{ env.ALLURE_DIR }}"
```

Comment:
Aren't we in `${{ github.workspace }}` already `ALLURE_DIR: ${{ github.workspace }}/allure-results`

Will have to double check the artifacts to see if the tar paths are concise.

Reply:
AFAIK no, we are in `${{ github.workspace }}/src` since we checkout two different repos and specify the default working-directory

Reply:
Can we vendor the test recipe instead of pulling it from an archived repository? Seems like a recipe for tears.

Reply:
Possibly but this PR is already pretty large and longrunning, can we do that in a followup PR?

Reply:
probably not, it's referred to in several tests as a cloned git repo

Reply:
we can perhaps change it to be a submodule?